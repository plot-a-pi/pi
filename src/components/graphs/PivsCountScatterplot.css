.container {
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  width: 85vw;
}

.svg {
  background: #212E59;
  display: block;
  width: 100%;
  height: 100%;
  color: whitesmoke;
}


