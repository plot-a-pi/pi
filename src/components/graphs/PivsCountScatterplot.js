/* eslint-disable babel/no-invalid-this */
import React, { useRef, useEffect, useState } from 'react';
import { select } from 'd3';
import { PropTypes } from 'prop-types';
import Styles from './PivsCountScatterplot.css';
// enables functionality in additional browsers
import ResizeObserver from 'resize-observer-polyfill';
import CSVButton from '../common/CSVButton';
import { makePivsCountScatterplot } from '../../d3/helpers';

const useResizeObserver = ref => {
  const [dimensions, setDimensions] = useState(null);

  useEffect(() => {
    const observeTarget = ref.current;
    const resizeObserver = new ResizeObserver(entries => {
      entries.forEach(entry => setDimensions(entry.contentRect));
    });
    resizeObserver.observe(observeTarget);
    return () => {
      resizeObserver.unobserve(observeTarget);
    };
  }, [ref]);
  return dimensions;
};

const PivsCountScatterplot = ({ data, title, xLabel, yLabel }) => {
  const svgRef = useRef(null);
  const wrapperRef = useRef(null);
  const dimensions = useResizeObserver(wrapperRef);
  
  useEffect(() => {
    const svg = select(svgRef.current);
    const wrapper = select(wrapperRef.current);

    const { width } = dimensions || wrapperRef.current.getBoundingClientRect();
    if(!dimensions) return;
    
    wrapper.style('height', `${0.55 * width}px`);
    
    makePivsCountScatterplot(svg, data, width);

  }, [data, dimensions]);

  return (
    <div className={Styles.container} ref={wrapperRef}>
      <svg className={Styles.svg} ref={svgRef}>
        <text className={'title'} fill='whitesmoke'>{title}</text>
        <g className={'x-axis'}></g>
        <text className={'x-label'} fill='whitesmoke'>{xLabel}</text>
        <g className={'y-axis'}></g>
        <text className={'y-label'} fill='whitesmoke'>{yLabel}</text>  
        <line className={'line'}></line>
      </svg>
      <div className={'csvButton'}>
        <CSVButton header1={xLabel} header2={yLabel} data={data} />
      </div>
    </div>
  );
};

PivsCountScatterplot.propTypes = {
  data: PropTypes.array.isRequired,
  title: PropTypes.string.isRequired,
  xLabel: PropTypes.string.isRequired,
  yLabel: PropTypes.string.isRequired
};

export default PivsCountScatterplot;
