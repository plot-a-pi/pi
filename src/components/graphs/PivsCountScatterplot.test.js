import React from 'react';
import { shallow } from 'enzyme';
import PivsCountScatterplot from './PivsCountScatterplot';

describe('PivsCountScatterplot component', () => {
  it('renders PivsCountScatterplot', () => {
    const wrapper = shallow(<PivsCountScatterplot data={[[1, 2]]} title={'test title'} xLabel={'test x-label'} yLabel={'test y-label'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
