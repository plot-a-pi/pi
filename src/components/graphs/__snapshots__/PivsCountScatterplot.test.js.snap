// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PivsCountScatterplot component renders PivsCountScatterplot 1`] = `
<div
  className="container"
>
  <svg
    className="svg"
  >
    <text
      className="title"
      fill="whitesmoke"
    >
      test title
    </text>
    <g
      className="x-axis"
    />
    <text
      className="x-label"
      fill="whitesmoke"
    >
      test x-label
    </text>
    <g
      className="y-axis"
    />
    <text
      className="y-label"
      fill="whitesmoke"
    >
      test y-label
    </text>
    <line
      className="line"
    />
  </svg>
  <CSVButton
    data={
      Array [
        Array [
          1,
          2,
        ],
      ]
    }
    header1="test x-label"
    header2="test y-label"
  />
</div>
`;
